### PR TITLE
Add introspection models

### PR DESCRIPTION
**Pull request by gildub**
Models and Collection for Introspection Rules

### CODE REVIEW

**Comment by seanhandley** [comment]
👍

File: lib/fog/openstack/models/introspection/rules_collection.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    class OpenStack
+      class RulesCollection < Fog::OpenStack::Collection
+        model Fog::Introspection::OpenStack::Rules
+
+        def all(_options = {})
+          load_response(service.list_rules, 'rules')
+        end
+
+        def summary(_options = {})
+          load_response(service.list_rules, 'rules')
```
Comment (Ladas) [inline lib/fog/openstack/models/introspection/rules_collection.rb]: you can omit this, when it's the same as all. Summary delegates to all in the superclass

File: lib/fog/openstack/models/introspection/rules_collection.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        end
+
+        def get(uuid)
+          data = service.get_rules(uuid).body
+          new(data)
+        rescue Fog::Introspection::OpenStack::NotFound
+          nil
+        end
+
+        def destroy(uuid)
```
Comment (Ladas) [inline lib/fog/openstack/models/introspection/rules_collection.rb]: just a note: we might just define this method on the superclass, seems like it's generic enough

**Comment by Ladas** [comment]
@gildub just a minor nit, otherwise LGTM :+1:

File: lib/fog/openstack/models/introspection/rules_collection.rb (hunk 1 above)
Comment (gildub) [inline lib/fog/openstack/models/introspection/rules_collection.rb]: I missed that, thanks!

File: lib/fog/openstack/models/introspection/rules_collection.rb (hunk 2 above)
Comment (gildub) [inline lib/fog/openstack/models/introspection/rules_collection.rb]: I understand the idea but I'm not following you 'cause the superclass only has method place holders. Are you willing to change that? Nevertheless if we change the superclass, I'd rather do that on a separate patch.

Comment (Ladas) [inline lib/fog/openstack/models/introspection/rules_collection.rb]: sure, this was a side note, so separate patch is ok. Yeah the superclass could hold a common functionality. Less code in the specific classes will be better. :-)

**Comment by Ladas** [comment]
looks great now :+1:
